docs/reference/builder.md

```diff
@@ -954,6 +954,125 @@ For example:
 The output of the final `pwd` command in this `Dockerfile` would be
 `/path/$DIRNAME`
 
+## ARG
+
+    ARG <name>=[<default value>]
+
+The `ARG` instruction defines a variable that users can pass at build-time to
+the builder with the `docker build` command using the `--build-arg
+<varname>=<value>` flag. If a user specifies a `--build-arg` but it is not
```

Comment:
s/`--build-arg but it is not used by the builder`/`a build argument that was not defined in the image`